GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1551 @@
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("Acquiring a token failed with the following error: {0}",
+                    Program.ExtractErrorMessage(e));
+                //TODO: Implement retry and back-off logic per the guidance given here:http://msdn.microsoft.com/en-us/library/dn168916.aspx
+                Console.ReadKey();
+                return;
+            }
+
+            #endregion
+
+            User newUser = new User();
+            Application newApp = new Application();
+            IDomain newDomain = new Domain();
+            Group newGroup = new Group();
+            ExtensionProperty newExtension = new ExtensionProperty();
+            ServicePrincipal newServicePrincipal = new ServicePrincipal();
+            OAuth2PermissionGrant newPermissionGrant = new OAuth2PermissionGrant();
+
+            try
+            {
+                Console.WriteLine("\nStarting user-mode requests...");
+                Console.WriteLine("\n=============================\n\n");
+
+
+                ITenantDetail tenantDetail = await GetTenantDetails(client);
+                User signedInUser = await GetSignedInUser(client);
+                await UpdateSignedInUsersPhoto(signedInUser);
+
+                Console.WriteLine("\nSearching for any user based on UPN, DisplayName, First or Last Name");
+                Console.WriteLine("\nPlease enter the user's name you are looking for:");
+
+                string searchString = Console.ReadLine();
+                await PeoplePickerExample(client, searchString);
+                await PrintUsersManager(signedInUser);
+                await PrintUsersGroupsAndRoles(signedInUser);
+
+                newUser = await CreateNewUser(client,
+                    tenantDetail.VerifiedDomains.First(x => x.@default.HasValue && x.@default.Value));
+                await UpdateNewUser(newUser);
+                await ResetUserPassword(newUser);
+                await AssignManager(newUser, signedInUser);
+                await UserGroupOperations(client, newUser);
+
+                newGroup = await CreateNewGroup(client);
+                await PrintAllRoles(client, searchString);
+                await PrintServicePrincipals(client);
+                await PrintApplications(client);
+
+                string graphAppObjectId = await GetAppObjectId(client, Constants.GraphServiceObjectId);
+                newApp = await CreateNewApplication(client, newUser);
+                newServicePrincipal = await CreateServicePrincipal(client, newApp);
+                string extName = "linkedInUserId";
+                newExtension = await CreateSchemaExtensions(client, newApp, extName);
+                await ManipulateExtensionProperty(newExtension, newUser);
+                PrintExtensionProperty(newUser, newExtension);
+
+                await AssignAppRole(client, newApp, newServicePrincipal);
+                await PrintDevicesAndOwners(client);

```

Comment:
>await PrintDevicesAndOwners(client); [](start = 16, length = 36)

Move down just before PrintAllDomains(client) so that all the permissions related methods are together.